src/library_syscall.js

```diff
@@ -159,16 +159,27 @@ var SyscallsLibrary = {
 #else
     // MEMFS filesystem disabled lite handling of stdout and stderr:
     buffers: [null, [], []], // 1 => stdout, 2 => stderr
-    printChar: function(stream, curr) {
+    printChar: function(stream, ch) {
       var buffer = SYSCALLS.buffers[stream];
 #if ASSERTIONS
       assert(buffer);
 #endif
-      if (curr === 0 || curr === {{{ charCode('\n') }}}) {
-        (stream === 1 ? out : err)(UTF8ArrayToString(buffer, 0));
+      if (ch === 0 || ch === {{{ charCode('\n') }}}) {
+        var str;
+        if (!buffer.hasUtf) str = String.fromCharCode.apply(null, buffer);
```

Comment:
nice, I did not know it can work on multiple inputs...

Reply:
Yeah, with up to 512k items IIRC, not sure if that is an issue... i'm not expecting `buffer` to be that long, but i may be wrong.

Reply:
Just tested this, and under Firefox 73 it's about ~500k while on Chorme 80 it's just ~120k items.

If you do envision that could be a problem we shall revert/remove such `hasUtf` logic (because a loop to iterate the buffer will make the code even bigger, which is not ideal)

Regarding code size though, while using `TEXTDECODER == 2` we're now getting rid of the `UTF8ArrayToString()` decency, so not that weird i'd say.

Btw, fwiw/fyi, I also tried to go beyond than that by getting rid of the `SYSCALLS.printChar()` dependency, and using an standalone function instead:

https://github.com/meganz/ffmpeg.js/blob/d54eec208db1bf3a7cef69d1e1a4de6c3723e1ab/build/pre.js#L42-L53

```js
function dump(fd, ptr, len) {
    db[fd] += String.fromCharCode.apply(null, HEAPU8.subarray(ptr, ptr + len));

    if (db[fd].indexOf('\n') >= 0) {
        var s = db[fd].split('\n');
        for (var i = 0; i < s.length - 1; i++) {
            (fd == 1 ? out : err)(s[i]);
        }
        db[fd] = s[s.length -1];
    }
}
```

This one is benchmarked and much faster, and at least under a `MINIMAL_RUNTIME` build (didn't tested outside of it), it allows the closure-compiler/uglifyjs to additionally get rid of the whole `SYSCALLS` object dependency, plus the `UTF8ToString()` dependency that e.g. `SYSCALLS.getStr()` does requires (given at least for my use-case it's unused), and as result of all that the code size is nicely decreased further.

If you are more interested in that other version, i could replace the `fromCharCode` by the `UTF8Decoder` and adapt `fd_write` accordingly, but i will need some hint(s) on how to do that (given the code _transpilation_ that happens there, which i am not familiar with)

Reply:
Oh, those limits make me worry actually. In theory the data here might be very large. We should probably avoid that approach, just to be safe.

The more I think about this in general, I think focusing on code size matters more. I don't remember this being slow enough for people to notice, so optimizing it for speed seems premature.

Reply:
Alright, i do agree that even reverting `fromCharCode` and leaving the `TEXTDECODER == 2` addition (or the alternate version), code size will only be decreased depending on what is being compiled, and therefore it may not be worth some times (dunno)

so yeah... let's discard this PR, and sorry for the time waste 😅

Reply:
No worries, not all ideas work out! Thanks for investigating.